hircluster.c

```diff
@@ -890,10 +886,7 @@ dict *parse_cluster_slots(redisClusterContext *cc, redisReply *reply,
 
                     ret = cluster_slot_ref_node(slot, master);
                     if (ret != REDIS_OK) {
-                        __redisClusterSetError(
-                            cc, REDIS_ERR_OOM,
-                            "Slot ref node failed: out of memory.");
-                        goto error;
```

Comment:
cluster_slot_ref_node() is not doing any allocations, and oom is checked on arguments before.
Will change back to `goto error;`

Reply:
I was wrong, it does allocations, like when using listAddNodeTail.
Seem very unlikely to have wrong role aswell, going for oom as we started with.